### ISSUE TITLE
LDAP authentication failing

### ISSUE THREAD

**Issue by gathomas72** [Requarks/wiki#1643]
Attempting to setup AD Authentication via ldap.   

Where is the troubleshooting log file?

![wikijs](https://user-images.githubusercontent.com/49758763/77960026-34d49380-729d-11ea-8bae-d73bdcb7237f.JPG)

**Comment by Smankusors** [comment]
Why is it failing? Is there any error? Or is it just shows wrong password when attempting to login?

**Comment by gathomas72** [comment]
Invalid user name or password error.   I have no knowledge of wiki.js.  Simply trying to get this up and running for a dev so users can use ad creds to login.  I don't know where in node.js to look for the debug log files to gather more intel.

**Comment by Smankusors** [comment]
You already selecting the LDAP strategy when attempting to login... right?

Anyway, how do you run Wiki.js anyway? Docker? Manual installation?

**Comment by gathomas72** [comment]
Yes I'm selecting ldap.   This is installed on a windows box.

**Comment by Smankusors** [comment]
You can try to enable LDAP debugging flag, which you can read more about it [here](https://docs.requarks.io/en/auth/ldap#troubleshooting)

**Comment by zhouweiw** [comment]
5e845f66 conn=211159 op=0 BIND dn="cn=admin,dc=service,dc=sd" method=128
5e845f66 conn=211159 op=0 BIND dn="cn=admin,dc=service,dc=sd" mech=SIMPLE ssf=0
5e845f66 conn=211159 op=0 RESULT tag=97 err=0 text=
5e845f66 conn=211159 op=1 SRCH base="ou=wikijs,dc=service,dc=sd" scope=2 deref=0 filter="(uid=test1)"
5e845f66 conn=211159 op=1 SEARCH RESULT tag=101 err=0 nentries=1 text=
5e845f66 conn=211160 op=0 BIND dn="uid=test1,ou=wikijs,dc=service,dc=sd" method=128
5e845f66 conn=211160 op=0 BIND dn="uid=test1,ou=wikijs,dc=service,dc=sd" mech=SIMPLE ssf=0
5e845f66 conn=211160 op=0 RESULT tag=97 err=0 text=
5e845f66 conn=211159 op=2 UNBIND
5e845f66 conn=211159 fd=36 closed
This is openldap‘s log, There is no error, but wikijs show 'You are not authorized to login.'  when I login.

**Comment by Smankusors** [comment]
is user that trying to log on exist on the wiki? you might need to enable "allow self register" option on the strategy.

Btw, you haven't given us full LDAP configuration on your wiki. You just give us first half of the configuration

**Comment by zhouweiw** [comment]
Thanks for your response！








------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Antony Kurniawan"<notifications@github.com&gt;;
发送时间:&nbsp;2020年4月1日(星期三) 下午5:43
收件人:&nbsp;"Requarks/wiki"<wiki@noreply.github.com&gt;;
抄送:&nbsp;"龙吟九天"<516596129@qq.com&gt;;"Comment"<comment@noreply.github.com&gt;;
主题:&nbsp;Re: [Requarks/wiki] LDAP authentication failing (#1643)





 
is user that trying to log on exist on the wiki? you might need to enable "allow self register" option on the strategy.
 
Btw, you haven't given us full LDAP configuration on your wiki. You just give us first half of the configuration

**Comment by Smankusors** [comment]
is your problem resolved? If so, you can close this issue :)

**Comment by gathomas72** [comment]
This was fixed by a couple of different changes.   In the Admin bind changed the following. 

It didn't like the use of OU in the DN for the AD OU portion.   
CN=User Name,**CN**=AD OU and not CN=User,OU=AD OU  

Once that was changed the error went to: **Missing or invalid email address from profile.**

I added word "mail" minus the quotation marks to the Email Field Mapping and then ldap authentication began working.
